spec/js-api/value/color.d.ts.md

```diff
@@ -39,83 +239,757 @@ The [private `internal` field] refers to a Sass color.
 
-  * Set [`internal`] to the result of [`hsl(hue, saturation, lightness,
-    alpha)`].
+  * Return `this`.
 
-  [`hsl(hue, saturation, lightness, alpha)`]: ../../functions.md#hsl-and-hsla
+[parsing a channel value]: #parsing-a-channel-value
+[`internal`]: #internal
+[`rgb(red green blue / alpha)`]: ../../functions.md#rgb-and-rgba
 
-* Otherwise, if `options.whiteness` is set:
+```ts
+constructor(options: {
+  red: number | null;
+  green: number | null;
+  blue: number | null;
+  alpha?: number | null;
+  space?: 'rgb';
+});
+```
+
+##### HSL
+
+* Otherwise, if `constructionSpace` is "hsl":
 
-  * Let `hue` be a Sass number with a value of `options.hue`.
+  * If `options.alpha` is `null` and `options.space` is not set, emit a
+    deprecation warning named `null-alpha`.
 
-  * Let `whiteness` be a Sass number with a value of `options.whiteness`.
+  * Let `hue` be the result of [parsing a channel value] with value
+    `options.hue`.
 
-  * Let `blackness` be a Sass number with a value of `options.blackness`.
+  * Let `saturation` be the result of [parsing a channel value] with value
+    `options.saturation`.
 
-  * If `options.alpha` is set, let `alpha` be a Sass number with a value of
-    `options.alpha`. Otherwise, let `alpha` be `null`.
+  * Let `lightness` be the result of [parsing a channel value] with value
+    `options.lightness`.
 
-  * Set [`internal`] to the result of [`hwb(hue, whiteness, blackness, alpha)`].
+  * If `options.alpha` is not set, let `alpha` be `1`. Otherwise, let `alpha` be
+    the result of [parsing a channel value] with value `options.alpha`.
 
-  [`hwb(hue, whiteness, blackness, alpha)`]: ../../built-in-modules/color.md#hwb
+  * Set [`internal`] to the result of [`hsl(hue saturation lightness / alpha)`].
 
-* Return `this`.
+  * Return `this`.
+
+[`hsl(hue saturation lightness / alpha)`]: ../../functions.md#hsl-and-hsla
 
 ```ts
 constructor(options: {
-  red: number;
-  green: number;
-  blue: number;
-  alpha?: number;
+  hue: number | null;
+  saturation: number | null;
+  lightness: number | null;
+  alpha?: number | null;
+  space?: ColorSpaceHSL;
 });
+```
+
+##### HWB
+
+* Otherwise, if `constructionSpace` is "hwb":
+
+  * If `options.alpha` is `null` and `options.space` is not set, emit a
+    deprecation warning named `null-alpha`.
+
+  * Let `hue` be the result of [parsing a channel value] with value
+    `options.hue`.
+
+  * Let `whiteness` be the result of [parsing a channel value] with value
+    `options.whiteness`.
+
+  * Let `blackness` be the result of [parsing a channel value] with value
+    `options.blackness`.
+
+  * If `options.alpha` is not set, let `alpha` be `1`. Otherwise, let `alpha` be
+    the result of [parsing a channel value] with value `options.alpha`.
+
+  * Set [`internal`] to the result of [`hwb(hue whiteness blackness / alpha)`].
+
+  * Return `this`.
+
+[`hwb(hue whiteness blackness / alpha)`]: ../../../proposal/color-4-new-spaces.md#hwb-1
```

Comment:
These links will need to be updated once the CSS Color Level 4, New Color Spaces proposal is accepted and added to the spec.